File: MobileApp/DeviceFinder.Android/Utilities/CommonData.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;
using Android.Content.Res;

namespace DeviceFinder.Droid.Utilities
{
    public static class CommonData
    {
        public static HashSet<string> Names { get; private set; }

        // Amazon won't tell us what first names are valid, so we need to check locally
        public static void Init(AssetManager assets)
        {
            Task.Run(() =>
            {
                using (StreamReader sr = new StreamReader(assets.Open("names.txt")))
                {
                    string rawNamesContent = sr.ReadToEnd();
                    Names = rawNamesContent.Split('\n').ToHashSet();
                }
            });
        }
    }
}

File: AndroidApp/app/src/main/java/com/jamespfluger/devicefinder/models/CommonData.java
package com.jamespfluger.alexadevicefinder.models;

import android.content.res.AssetManager;
import android.os.AsyncTask;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.nio.charset.StandardCharsets;
import java.util.HashSet;

public final class CommonData {
    public static HashSet<String> names = new HashSet<>();

    public void initData(AssetManager assets) {
        new AsyncFileReader().execute(assets);
    }

    // Amazon won't tell us what first names are valid, so we need to check locally
    private static class AsyncFileReader extends AsyncTask<AssetManager, Void, HashSet<String>> {
        @Override
        protected HashSet<String> doInBackground(AssetManager... assets) {
            try (BufferedReader reader = new BufferedReader(new InputStreamReader(assets[0].open("names.txt"), StandardCharsets.UTF_8))) {
                String name;
                while ((name = reader.readLine()) != null) {
                    names.add(name);
                }
            } catch (IOException e) {
            }

            return names;
        }
    }
}
